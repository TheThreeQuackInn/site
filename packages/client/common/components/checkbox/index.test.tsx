import React from 'react';
import renderer from 'react-test-renderer';
import Checkbox from './';

describe('Checkbox', () => {
    it('renders correctly', () => {
        const onChange = jest.fn();
        const tree = renderer
            .create(<Checkbox checked={false} onChange={onChange} name="Test" label="Testing" />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
