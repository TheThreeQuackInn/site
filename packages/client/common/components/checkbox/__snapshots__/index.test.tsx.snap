// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders the checked state correctly 1`] = `
<label
  className="inline-flex items-center mr-2 test"
  htmlFor="test"
>
  <input
    checked={true}
    className="form-checkbox focus:outline-black dark:focus:outline-white"
    id="test"
    name="test"
    onChange={[MockFunction]}
    onKeyPress={[Function]}
    type="checkbox"
  />
  <span
    className="ml-1 text-gray-700 dark:text-white"
  >
    Test
  </span>
</label>
`;

exports[`Checkbox renders the unchecked state correctly 1`] = `
<label
  className="inline-flex items-center mr-2 test"
  htmlFor="test"
>
  <input
    checked={false}
    className="form-checkbox focus:outline-black dark:focus:outline-white"
    id="test"
    name="test"
    onChange={[MockFunction]}
    onKeyPress={[Function]}
    type="checkbox"
  />
  <span
    className="ml-1 text-gray-700 dark:text-white"
  >
    Test
  </span>
</label>
`;
